CONTRIBUTING.md

```diff
@@ -0,0 +1,58 @@
+# Contributing to the Terraform Helm Provider
+
+
+---
+
+**All communication on GitHub, the community forum, and other HashiCorp-provided communication channels is subject to [the HashiCorp community guidelines](https://www.hashicorp.com/community-guidelines).**
+
+This document provides guidance on our recommended practices for contribution. It covers what we're looking for in order to help set some expectations and help you get the most out of participation in this project. 
+
+To record a bug report, enhancement proposal, or give any other product feedback, please [open a GitHub issue](https://github.com/terraform-providers/terraform-provider-helm/issues/new/choose) using the most appropriate issue template. Please do fill in all of the information the issue templates request, because we've seen from experience that this will maximize the chance that we'll be able to act on your feedback.
+
+---
+
+<!-- MarkdownTOC autolink="true" -->
+
+- [Contributing Fixes](#contributing-fixes)
+- [Proposing a Change](#proposing-a-change)
+- [Maintainers](#maintainers)
+- [Pull Request Lifecycle](#pull-request-lifecycle)
+  - [Getting Your Pull Requests Merged Faster](#getting-your-pull-requests-merged-faster)
+
+<!-- /MarkdownTOC -->
+
+## Proposing a Change
+
+In order to be respectful of the time of community contributors, we aim to discuss potential changes in GitHub issues prior to implementation. That will allow us to give design feedback up front and set expectations about the scope of the change, and, for larger changes, how best to approach the work such that the team can review it and merge it along with other concurrent work.
+
+If the bug you wish to fix or enhancement you wish to implement isn't already covered by a GitHub issue that contains feedback from the team, please do start a discussion (either in [a new GitHub issue](https://github.com/terraform-providers/terraform-provider-helm/issues/new/choose) or an existing one, as appropriate) before you invest significant development time. If you mention your intent to implement the change described in your issue, the team can, as best as possible, prioritize including implementation-related feedback in the subsequent discussion.
```

Comment:
I've read this part a few times, but I'm having trouble understanding what it means. Could you break it down into simpler terms, or maybe walk me through a scenario?

`If you mention your intent to implement the change described in your issue, the team can, as best as possible, prioritize including implementation-related feedback in the subsequent discussion.`